refinery/data_set_manager/utils.py

```diff
@@ -503,7 +505,44 @@ def _index_annotated_nodes(node_type, study_uuid, assay_uuid=None,
     logger.info("%s nodes indexed in %s", str(counter), str(end - start))
 
 
-def generate_solr_params(params, assay_uuid):
+def generate_solr_params_for_user(params, user):
+    """Creates the encoded solr params limiting results to one user.
+    Keyword Argument
+        params -- python dict or QueryDict
+    Params/Solr Params
+        is_annotation - metadata
+        facet_count/facet - enables facet counts in query response, true/false
+        offset - paginate, offset response
+        limit/row - maximum number of documents
+        field_limit - set of fields to return
+        facet_field - specify a field which should be treated as a facet
+        facet_filter - adds params to facet fields&fqs for filtering on fields
+        facet_pivot - list of fields to pivot
+        sort - Ordering include field name, whitespace, & asc or desc.
+        fq - filter query
+     """
+
+    assay_uuids = []
+    datasets = get_objects_for_user(user, "core.read_dataset")
+    for dataset in datasets:
+        version_details = dataset.get_version_details()
```

Comment:
I agree in general, but I think that here, breaking out the checks line by line will require a more complicated structure than checking the whole set. That said, I might be looking at it wrong, so I'll go in the direction you suggest and we see if that's better.